### PR TITLE
XCM Doc Restructure: Adding pages

### PR DESCRIPTION
**Pull request by CrackTheCode016**
This PR marks the initial setup in the effort to clarify and elaborate the documentation centered around XCM.

**Issue Reference**: https://github.com/w3f/polkadot-wiki/issues/4541

The following pages are proposed in this PR: 

- Introduction to Cross Consensus Messaging (XCM)
- XCM FRAME Pallet Overview
- XCM Usecases & Examples
- XCM Transport Methods (XCMP, HRMP, VMP)
- XCM Virtual Machine (XCVM) & XCM Executor
- XCM Instructions & Register Specification

This PR will only cover basic info / moving around existing documentation to provide clarity and structure.  Subsequent Page-specific PRs will follow later.

### CODE REVIEW

File: docs/learn/learn-xcm.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 id: learn-xcm
-title: Cross-Consensus Message Format (XCM)
-sidebar_label: Cross-Consensus Message Format (XCM)
+title: Introduction to Cross-Consensus Message Format (XCM)
+sidebar_label: Introduction to Cross-Consensus Message Format (XCM)
```
Comment (profradha) [inline docs/learn/learn-xcm.md]: Will be too lengthy for the Sidebar

```suggestion
sidebar_label: Cross-Consensus Message Format (XCM)
```

File: docs/learn/learn-xcvm.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
+---
+id: learn-xcvm
+title: XCM Virtual Machine (XCVM) & XCM Executor
+sidebar_label: XCM Virtual Machine (XCVM) & XCM Executor
```
Comment (profradha) [inline docs/learn/learn-xcvm.md]: Shortening sidebar label. If you think XCM executor should be displayed in the side bar, feel free to ignore this commit.

```suggestion
sidebar_label: XCM Virtual Machine (XCVM)
```

File: docs/learn/learn-xcm-instructions.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
+---
+id: learn-xcm-instructions
+title: XCM Instructions & Register Specification
+sidebar_label: XCM Instructions & Register Specification
```
Comment (profradha) [inline docs/learn/learn-xcm-instructions.md]:
```suggestion
sidebar_label: XCM Instructions
```

File: docs/learn/learn-xcm-pallet.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
+---
+id: learn-xcm-pallet
+title: XCM FRAME Pallet Overview
+sidebar_label: XCM FRAME Pallet Overview
```
Comment (profradha) [inline docs/learn/learn-xcm-pallet.md]:
```suggestion
sidebar_label: XCM Pallet
```

File: docs/learn/learn-xcm-transport.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
+---
+id: learn-xcm-transport
+title: XCM Transport Methods (XCMP, HRMP, VMP)
+sidebar_label: XCM Transport Methods (XCMP, HRMP, VMP)
```
Comment (profradha) [inline docs/learn/learn-xcm-transport.md]:
```suggestion
sidebar_label: XCM Transport
```

File: docs/learn/learn-xcm-usecases.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
+---
+id: learn-xcm-usecases
+title: XCM Usecases & Examples
+sidebar_label: XCM Usecases & Examples
```
Comment (profradha) [inline docs/learn/learn-xcm-usecases.md]:
```suggestion
sidebar_label: XCM Usecases
```

File: docs/learn/learn-xcm-pallet.md [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+description: Learn how the XCM pallet works, and its place in the ecosystem.
+keywords: [cross-consensus, XCM, XCMP, interoperability, communication]
+slug: ../learn-xcm-pallet
+---
+
+The XCM pallet ([`pallet-xcm`](https://github.com/paritytech/polkadot/tree/master/xcm/pallet-xcm))
+provides a developer-friendly interface for more of the common XCM messages.
```
Comment (profradha) [inline docs/learn/learn-xcm-pallet.md]:
```suggestion
provides a developer-friendly interface for most of the common XCM messages.
```

File: docs/learn/learn-xcm-pallet.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+This pallet provides default implementations for many traits required by `XcmConfig`, which is
+highly useful for configuration.
+
+It defines a set of ten extrinsics that can be utilized to build XCVM programs, either to target the
```
Comment (profradha) [inline docs/learn/learn-xcm-pallet.md]: Keep the wiki writeup generic. We won't be able to spot this if they add a 11th extrinsic :)

```suggestion
It defines a set of extrinsics that can be utilized to build XCVM programs, either to target the
```

File: docs/learn/learn-xcm-transport.md [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+1. XCMP (Cross-Chain Message Passing)
+2. Horizontal Relay-routed Message Passing (HRMP/XCMP-lite)
+3. VMP (Vertical Message Passing)
+
+### XCMP (Cross-Chain Message Passing)
+
+:::caution
+
+XCMP is currently under development and most cross-chain message passing uses HRMP channels for the
+time being.
```
Comment (profradha) [inline docs/learn/learn-xcm-transport.md]:
```suggestion
XCMP is currently under development, and most of the cross-chain messages pass through HRMP channels for the
time being.
```

File: docs/learn/learn-xcm-usecases.md [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+description: Learn how XCM can be used in cross consensus interactions.
+keywords: [cross-consensus, XCM, XCMP, interoperability, communication]
+slug: ../learn-xcm-usecases
+---
+
+XCM has a multitude of various use cases. While the wiki covers some of the more commonplace
```
Comment (profradha) [inline docs/learn/learn-xcm-usecases.md]:
```suggestion
XCM has a multitude of use cases. While the wiki covers some of the key commonplace
```

**Review by DrW3RK — APPROVED** [review]
Minor suggestions

File: docs/learn/learn-xcvm.md (hunk 2 above)
Comment (CrackTheCode016) [reply docs/learn/learn-xcvm.md]: No, it makes sense for the sidebar label - XCVM kind of implies the executor anyway 👍

**Comment by CrackTheCode016** [comment]
Thank you! Might have to change some of the sidebar naming schemes in the future, i.e, since we're in the XCM section, maybe prefacing everything with XCM is unnecessary. Not relevant for this PR, though.

**Pull request merged**